assignment-01-cat/Cat.cpp

```diff
@@ -1,15 +1,69 @@
 #include "Cat.h"
+#include <cstring>
 
-Cat::Cat(const char* _name) {
+Cat::Cat(const char* _name)
+{
     // TODO: this->name에 동적할당을 하고 _name을 복사해야 합니다.
+    this->name = new char[strlen(_name)+1];
+    strcpy(name, _name);
     // TODO: this->age는 -1으로 초기화해야 합니다.
+    this->age = -1;
     // TODO: this->sex는 false로 초기화해야 합니다.
+    this->sex = false;
 }
 
-Cat::Cat(const char* _name, int _age) {
+Cat::Cat(const char* _name, int _age)
+{
     // TODO: this->name에 동적할당을 하고 _name을 복사해야 합니다.
-    // TODO: this->age에 _age로 초기화해야 합니다.
+    this->name = new char[strlen(_name)+1];
+    strcpy(name, _name);
+    // TODO: this->age에 _age로 초기화야 합니다.
+    this->age = _age;
     // TODO: this->sex는 false로 초기화해야 합니다.
+    this->sex = false;
 }
 
-// TODO: 나머지 모두 구현하세요.
+Cat::Cat(const char* _name, int _age, bool _sex)
+{
+    this->name = new char[strlen(_name)+1];
+    strcpy(name, _name);
+    this->age = age;
+    this->sex = false;
+}
+
+Cat::Cat(const Cat& cat)
+{
+    name = new char(*cat.name);
+    
+}
```

Comment:
```suggestion
Cat::Cat(const Cat& cat): Cat(cat.name, cat.age, cat.sex) {}
```
코드가 무얼 하는 지 모르겠네...👶🏻
복사대입 연산자는 전달받은 cat과 동일한 값을 가지는 객체를 만들어주는 생성자입니다!...
마찬가지로 이런식으로 위에서 이미 정의한 생성자를 호출해주면 간단히 해결됩니다...